src/main/java/com/Nearsoft/referrals/service/impl/StorageServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.Nearsoft.referrals.service.impl;
+
+import com.Nearsoft.referrals.service.StorageService;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+
+@Service
+public class StorageServiceImpl implements StorageService {
```

Comment:
Service classes must be state less... What happens if two requests use this service at the same time? 

You can create a StorageData class containing rootLocation and fileName instead, and StorageServiceImpl::store could return an instance of it.